buzsaki_lab_to_nwb/fujisawa_code/fujisawamiscdatainterface.py

```diff
@@ -0,0 +1,108 @@
+import numpy as np
+from pathlib import Path
+from scipy.io import loadmat
+import warnings
+
+from nwb_conversion_tools.basedatainterface import BaseDataInterface
+from pynwb import NWBFile
+from pynwb.file import TimeIntervals
+from pynwb.behavior import SpatialSeries, Position
+from hdmf.backends.hdf5.h5_utils import H5DataIO
+
+from ..neuroscope import get_events, check_module
+
+
+class FujisawaMiscInterface(BaseDataInterface):
+    """Primary data interface for miscellaneous aspects of the FujisawaS dataset."""
+
+    @classmethod
+    def get_source_schema(cls):
+        return dict(properties=dict(folder_path="string"))
+
+    def run_conversion(
+        self,
+        nwbfile: NWBFile,
+        metadata_dict: dict,
+        stub_test: bool = False,
+     ):
+        session_path = Path(self.input_args['folder_path'])
+        session_id = session_path.name
+
+        # Stimuli
+        [nwbfile.add_stimulus(x) for x in get_events(session_path)]
+
+        # States
+        sleep_state_fpath = session_path / f"{session_id}.SleepState.states.mat"
+        # label renaming
+        state_label_names = dict(WAKEstate="Awake", NREMstate="Non-REM", REMstate="REM")
+        if os.path.isfile(sleep_state_fpath):
+            matin = loadmat(sleep_state_fpath)['SleepState']['ints'][0][0]
+
+            table = TimeIntervals(name='states', description="Sleep states of animal.")
+            table.add_column(name='label', description="Sleep state.")
+
+            data = []
+            for name in matin.dtype.names:
+                for row in matin[name][0][0]:
+                    data.append(dict(start_time=row[0], stop_time=row[1], label=state_label_names[name]))
+            [table.add_row(**row) for row in sorted(data, key=lambda x: x['start_time'])]
+            check_module(nwbfile, 'behavior', "contains behavioral data").add_data_interface(table)
+
+        # Position
+        pos_filepath = Path(session_path) / f"{session_id}.position.behavior.mat"
+        pos_mat = loadmat(str(pos_filepath.absolute()))
+        starting_time = float(pos_mat['position']['timestamps'][0][0][0])  # confirmed to be a regularly sampled series
+        rate = float(pos_mat['position']['timestamps'][0][0][1]) - starting_time
+        unit = pos_mat['position']['units'][0][0][0]
+        if unit == 'm':
+            conversion = 1.0
+        else:
+            warnings.warn(f"Spatial unit ({unit}) not listed in meters; setting conversion to nan.")
+            conversion = np.nan
+        pos_xy = []
+        for dim in ['x', 'y']:
+            pos_xy.append(pos_mat['position']['position'][0][0][dim][0][0])
+        pos_data = [
+            [x[0], y[0]] for x, y in zip(pos_xy[0], pos_xy[1])
+        ]
+        linearized_data = [[lin[0]] for lin in pos_mat['position']['position'][0][0]['lin'][0][0]]
+
+        label = pos_mat['position']['behaviorinfo'][0][0]['MazeType'][0][0][0].replace(" ", "")
+        pos_obj = Position(name=f"{label}Position")
```

Comment:
shouldn't this type of information be extracted with `get_metadata()` and inside `run_conversion()` be taken from the passed metadata? This would allow the user to rename these fields